Gold Buy Dip/gold_buy_dip_strategy_FIXED.py

```
@@ -0,0 +1,521 @@
+                
+            trigger = self.check_percentage_trigger()
+            if trigger:
+                # Additional filter: avoid same direction trades too close together
+                if (self.state.last_grid_close_direction and 
+                    trigger.value == self.state.last_grid_close_direction and
+                    self.state.last_grid_close_price > 0):
+                    
+                    current_price = candle.close
+                    if self.state.last_grid_close_direction == "BUY":
+                        if current_price >= self.state.last_grid_close_price * (1 - self.state.min_price_move_for_new_grid / 100):
+                            logger.debug(f"Same direction filter: BUY too close to last close")
```

Ruff (0.13.3)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)